apps/prairielearn/src/lib/submissions-upload.ts

```diff
@@ -300,3 +242,368 @@ export async function uploadSubmissions(
+        if (!rubric_item) {
+          continue;
+        }
+        applied_rubric_item_ids.push(rubric_item.id);
+      }
+    }
+    manual_rubric_data = {
+      rubric_id: assessmentQuestion.manual_rubric_id,
+      applied_rubric_items: applied_rubric_item_ids.map((id) => ({
+        rubric_item_id: id,
+      })),
+      adjust_points: row['Rubric Grading']?.adjust_points ?? null,
+    };
+  }
+
+  await updateInstanceQuestionScore(
+    assessment_id,
+    instance_question_id,
+    submission_id,
+    null,
+    {
+      manual_score_perc: null,
+      manual_points: run(() => {
+        if (assessmentQuestion.manual_rubric_id) {
+          return row['Rubric Grading']?.computed_points ?? null;
+        } else {
+          return row['Manual points'];
+        }
+      }),
+      auto_score_perc: null,
+      auto_points: row['Auto points'] ?? null,
+      feedback: row.Feedback,
+      manual_rubric_data,
+    },
+    authn_user_id,
+  );
+}
+
+async function processGroupSubmissionRow(record: any, context: ProcessingContext): Promise<void> {
+  const row = GroupSubmissionCsvRowSchema.parse(record);
+  const {
+    assessment_id,
+    course_id,
+    course_instance_id,
+    authn_user_id,
+    maxPoints,
+    selectUser,
+    selectQuestion,
+    selectAssessmentQuestion,
+    getOrInsertAssessmentInstance,
+    getOrInsertInstanceQuestion,
+    getOrInsertVariant,
+  } = context;
+
+  const question = await selectQuestion(row.Question);
+  const assessmentQuestion = await selectAssessmentQuestion(question.id);
+
+  // Create or get group and add all users to it
+  const groupName = row['Group name'];
+  const usernames = row.Usernames;
+
+  if (usernames.length === 0) {
+    throw new Error(`Group "${groupName}" has no usernames`);
+  }
+
+  // Create users for all group members concurrently
+  const users = await Promise.all(usernames.map((uid) => selectUser(uid)));
+
+  // Get the group ID - either existing or newly created
+  // First, try to find existing group
+  let group_id = await sqldb.queryOptionalRow(
+    sql.select_group_by_name,
+    { group_name: groupName, assessment_id },
+    IdSchema,
+  );
+
+  // If group doesn't exist, create it and add all members
+  if (!group_id) {
+    // Get the group config for this assessment
+    const groupConfig = await sqldb.queryRow(
+      sql.select_group_config,
+      { assessment_id },
+      z.object({
+        id: z.string(),
+        course_instance_id: z.string(),
+      }),
+    );
+
+    // Create the group
+    group_id = await sqldb.queryRow(
+      sql.create_group,
+      { assessment_id, authn_user_id, group_name: groupName },
+      IdSchema,
+    );
+
+    // Add all users to the group
+    for (const user of users) {
+      await sqldb.execute(sql.insert_group_user, {
+        group_id,
+        user_id: user.user_id,
+        group_config_id: groupConfig.id,
+        authn_user_id,
+      });
+    }
+  }
```

Comment:
There are some functions that sorta already do this in `apps/prairielearn/src/lib/groups.ts`. Can we use those to avoid code duplication?